src/Filesystem/Folder.php

```diff
@@ -194,16 +194,32 @@ public function read($sort = true, $exceptions = false, $fullPath = false)
             if ($fullPath) {
                 $name = $item->getPathname();
             }
+
             if ($item->isDir()) {
-                $dirs[] = $name;
+                $dirs[$item->getCTime()][] = $name;
             } else {
-                $files[] = $name;
+                $files[$item->getCTime()][] = $name;
```

Comment:
@chinpei215 hey :)
$item->getCTime() is always called but the `korder` call is called depending on the sort chosen.

Apart from that, imo I would consider correct:
```
ctime name
100   a.txt
100   c.txt
101   b.txt
```
and
```
ctime name
100   c.txt
100   a.txt
101   b.txt
```

Because `a.txt` and `c.txt` has the same `ctime` and we are not sorting by time and name, only but time. I created another test that accomplish it and change some gaps.